rust/generated_parser/src/early_errors.rs

```diff
@@ -40,6 +41,227 @@ pub trait ParameterEarlyErrorsContext<'alloc> {
+        //
+        // IdentifierReference[Yield, Await] : Identifier
+        //
+        // BindingIdentifier[Yield, Await] : Identifier
+        //
+        // LabelIdentifier[Yield, Await] : Identifier
+        //
+        // * It is a Syntax Error if this production has a [Yield] parameter
+        //   and StringValue of Identifier is "yield".
+        return Err(ParseError::NotImplemented("[Yield] parameter"));
+
+        // IdentifierReference : yield
+        //
+        // BindingIdentifier : yield
+        //
+        // LabelIdentifier : yield
+        //
+        // * It is a Syntax Error if the code matched by this production is
+        //   contained in strict mode code.
+        //
+        // and
+        //
+        // Identifier : IdentifierName but not ReservedWord
+        //
+        // * It is a Syntax Error if this phrase is contained in strict mode
+        //   code and the StringValue of IdentifierName is: "implements",
+        //   "interface", "let", "package", "private", "protected", "public",
+        //   "static", or "yield".
+        //
+        // if self.is_strict()? {
+        //     return Err(ParseError::InvalidIdentifier(
+        //         name.clone(),
+        //         offset,
+        //     ));
+        // }
+        //
+        // Ok(())
+    }
+
+    fn check_await_common(&self, _offset: usize) -> EarlyErrorsResult<'alloc> {
+        // Static Semantics: Early Errors
+        // https://tc39.es/ecma262/#sec-identifiers-static-semantics-early-errors
+        //
+        // IdentifierReference[Yield, Await] : Identifier
+        //
+        // BindingIdentifier[Yield, Await] : Identifier
+        //
+        // LabelIdentifier[Yield, Await] : Identifier
+        //
+        // * It is a Syntax Error if this production has an [Await] parameter
+        //   and StringValue of Identifier is "await".
+        return Err(ParseError::NotImplemented("[Await] parameter"));
+
+        // IdentifierReference : await
+        //
+        // BindingIdentifier : await
+        //
+        // LabelIdentifier : await
+        //
+        // * It is a Syntax Error if the goal symbol of the syntactic
+        //   grammar is Module.
+        //
+        // and
+        //
+        // Identifier : IdentifierName but not ReservedWord
+        //
+        // * It is a Syntax Error if the goal symbol of the syntactic grammar
+        //   is Module and the StringValue of IdentifierName is "await".
+        //
+        // if self.is_module()? {
+        //     return Err(ParseError::InvalidIdentifier(
+        //         name.clone(),
+        //         offset,
+        //     ));
+        // }
+        //
+        // Ok(())
+    }
+
+    fn check_identifier(&self, name: Name<'alloc>, offset: usize) -> EarlyErrorsResult<'alloc> {
+        match name {
+            "implements" | "interface" | "let" | "package" | "private" | "protected" | "public"
```

Comment:
Hmm. Most identifiers will be checked twice, right? This seems like it will make parsing slower, but OK.

Reply:
from the generated code, it looks like it's doing comparison for each string.
So, separating match into 2 part won't be something we have to worry about.

we could improve this later, by doing something similar to `ReservedWordsGenerated` in C++ frontend.

Reply:
oh, I misunderstood what you mean.
indeed, it's checked both in lexer and here.
we could embed some information about not-being contextual keyword and avoid check here.